yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -760,7 +756,13 @@ object Client extends Logging {
 
   /**
    * Populate the classpath entry in the given environment map.
-   * This includes the user jar, Spark jar, and any extra application jars.
+   *
+   * User jars are generally not added to the JVM's classpath; those are handled by the AM and
```

Comment:
Are we going to extra pains to support spark.yarn.user.classpath.first here?  If so, I think we should just drop support for it, because it was never documented in the first place.